src/services/user-config-service.ts

```diff
@@ -0,0 +1,120 @@
+import { getLogger } from "config/logger";
+import { RepoSyncState } from "models/reposyncstate";
+import { GitHubInstallationClient } from "../github/client/github-installation-client";
+import { Config } from "interfaces/common";
+import YAML from "yaml";
+import { InstallationId } from "../github/client/installation-id";
+import { Subscription } from "models/subscription";
+
+const USER_CONFIG_FILE = ".jira/config.yml";
+const logger = getLogger("services.user-config");
+const MAX_PATTERNS_PER_ENVIRONMENT = 10;
+
+/**
+ * Checks whether a list of modified files contains the config file. If yes, reads that config file
+ * from the GitHub repository, parses it, and stores the config against the given repository
+ * in the database.
+ *
+ * This function is meant to be called whenever there is a change in the repository so we can check
+ * if the config file has changed.
+ *
+ * @param subscription the subscription to which the repository belongs.
+ * @param repositoryId the ID of the repository.
+ * @param githubInstallationId the ID of the installation to which the repository belongs.
+ * @param modifiedFiles list of modified files (added, modified, or removed). The config will only be updated if this list contains
+ * the config file.
+ */
+export const updateRepoConfig = async (
+	subscription: Subscription,
```

Comment:
Yeah, technically, we would only need to pass around the `subscriptionId` instead of the whole `subscription`. But the database APIs all take the subscription object as an input and it's a very central object that is pretty much available everywhere, anyways. 

Also, this way it's typed instead of just passing around a generic integer `subscriptionId` :).